crypto/rand/rand_unix.c

```diff
@@ -325,6 +334,52 @@ static struct random_device {
 } random_devices[OSSL_NELEM(random_device_paths)];
 static int keep_random_devices_open = 1;
 
+#if defined(__linux)
+static void *shm_addr;
+
+# if !defined(FIPS_MODE)
+static void cleanup_shm(void)
+{
+    shmdt(shm_addr);
+}
+# endif
+#endif
+
+/*
+ * Ensure that the system randomness source has been adequately seeded.
+ * This is done by having the first start of libcrypto, read one bytes from
+ * /dev/random.  Subsequent starts of the library and later reseedings do not
+ * need to do this.
+ */
+static void wait_random_seeded(void)
+{
+#if defined(__linux)
+    static const char random_path[] = "/dev/random";
+    static int seeded = 0;
+    int shm_id, fd;
+    char c;
+
+    if (!seeded) {
+        if ((shm_id = shmget(OSSL_SHM_SEEDED, 1, 0)) == -1) {
+            if ((fd = open(random_path, O_RDONLY)) != -1) {
+                if (read(fd, &c, 1) == 1)
+                    shm_id = shmget(OSSL_SHM_SEEDED, 1, IPC_CREAT);
```

Comment:
You need to set permission bits.

Reply:
The shared memory is neither read nor written.  Mode 000 seems reasonable in this case.

Reply:
My mistake, permissions are required here so that the later `shmat(2)` call can map the region.